### PR TITLE
Ocaml 405 fix

### PR DESCRIPTION
**Pull request by rijesha**
Various ocaml fixes for ocaml build warnings.

Warning 52 was fixed by running the exception for all the possible cases. Most of these were for fairly trivial errors ie:
Failure ("int_of_string")  changed to Failure _

Code was tested with ocaml 4.01, 4.02, 4.04 and 4.05.

See issue #2140

### CODE REVIEW

File: sw/lib/ocaml/compat.ml [hunk 1]
```diff
@@ -75,3 +69,39 @@ let bytes_set = fun s n c->
 
 let bytes_iter = fun f s->
   BYTES.iter f s
+
+IFDEF OCAML_V404 THEN
```
Comment (gautierhattenberger) [inline sw/lib/ocaml/compat.ml]: Since uppercase/lowercase/capitalize functions from Bytes and String modules seems to have the same effect, I think we can make something simpler like:
```
IFDEF OCAML_V404 THEN
let lowercase_ascii = BYTES.lowercase_ascii
let uppercase_ascii = BYTES.uppercase_ascii
let capitalize_ascii = BYTES.capitalize_ascii
ELSE
let lowercase_ascii = BYTES.lowercase
let uppercase_ascii = BYTES.uppercase
let capitalize_ascii = BYTES.capitalize
END
```
This way the functions from `Compat` will match the new standards and `BYTES` will pick the correct module.

File: sw/ground_segment/cockpit/page_settings.ml [hunk 2]
```diff
@@ -55,7 +55,7 @@ object
           let v = float_of_string s in
           last_known_value <- Some v;
           set_default v
-        with Failure "float_of_string" -> ()
+        with Failure _ -> ()
```
Comment (gautierhattenberger) [inline sw/ground_segment/cockpit/page_settings.ml]: why did you remove all these filtering ?

**Comment by gautierhattenberger** [comment]
I merged the corresponding pprzlink PR, please update for the correct revision

**Comment by rijesha** [comment]
I think this branch is already set for the latest pprzlink commit.

**Comment by gautierhattenberger** [comment]
since I did a merge and rebase, latest commit is now https://github.com/paparazzi/pprzlink/commit/25b7024ad46964b4c99c998b75eca6db34ceab5b
